src/jit/utils.cpp

```diff
@@ -1544,3 +1544,64 @@ unsigned            CountDigits(unsigned num, unsigned base /* = 10 */)
 }
 
 #endif // DEBUG
+
+
+double FloatingPointUtils::convertUInt64ToDouble(unsigned __int64 uIntVal) {
+    __int64 s64 = uIntVal; 
+    double d;
+    if (s64 < 0) {
+#if defined(_TARGET_XARCH_) && !defined(_MSC_VER)
```

Comment:
We also need a comment in CodegenXarch.cpp genIntToFloatCast line 7500 that says that the sequence that we generate round incorrectly but is used for backwards compatibility.  And that if the sequence is changed in the future that the constant folding routine FloatingPointUtils::convertUInt64ToDouble also need to be updated.